testsupport/signup_request.go

```diff
@@ -305,7 +299,7 @@ func (r *SignupRequest) Execute(t *testing.T) *SignupRequest {
 		cleanup.AddCleanTasks(t, hostAwait.Client, r.userSignup)
 	}
 
-	return r
+	return r.userSignup, r.mur, r.space, r.token
 }
```

Comment:
I was more thinking about encapsulating the returned values into a new type:
```go 
...
	return SignupResult{
		UserSignup: userSignup,
		Mur:        mur,
		Space:      space,
		Token:      token,
	}
}

type SignupResult struct {
	UserSignup *toolchainv1alpha1.UserSignup
	MUR        *toolchainv1alpha1.MasterUserRecord
	Space      *toolchainv1alpha1.Space
	Token      string
}
```

so it can be then used:
```go
user1 := NewSignupRequest(awaitilities).
                ...
                Execute(t)
```
all these returned values are related to the specific UserSignup, so encapsulating them and using via the type 
```go
user1.Space
user1.MUR
...
```
would be OK and would make sense in the context of the test suites - that you reference to some specific provisioned user account and the resources provisioned as part of the account.

WDYT?

Reply:
It makes perfect sense to me! The code will be more clean. Thanks!